ambry-frontend/src/main/java/com/github/ambry/frontend/NamedBlobDeleteHandler.java

```diff
@@ -0,0 +1,109 @@
+   * @param namedBlobDb The {@link NamedBlobDb} to use.
+   * @param metrics The {@link FrontendMetrics} instance where metrics are recorded
+   * @param deleteBlobHandler The {@link DeleteBlobHandler}.
+   */
+  NamedBlobDeleteHandler(AccountAndContainerInjector accountAndContainerInjector, NamedBlobDb namedBlobDb,
+      FrontendMetrics metrics, DeleteBlobHandler deleteBlobHandler) {
+    this.accountAndContainerInjector = accountAndContainerInjector;
+    this.namedBlobDb = namedBlobDb;
+    this.metrics = metrics;
+    this.deleteBlobHandler = deleteBlobHandler;
+  }
+
+  /**
+   * Asynchronously handle named blob delete requests.
+   * @param restRequest the {@link RestRequest} that contains the request parameters and body.
+   * @param restResponseChannel the {@link RestResponseChannel} where headers should be set.
+   * @param callback the {@link Callback} to invoke when the response is ready (or if there is an exception).
+   * @throws RestServiceException
+   */
+  void handle(RestRequest restRequest, RestResponseChannel restResponseChannel, Callback<Void> callback)
+      throws RestServiceException {
+    RequestPath requestPath = getRequestPath(restRequest);
+    // Get blob id and reconstruct RequestPath
+    // this is the three-part named blob id, including account name, container name and the custom blob name.
+    String namedBlobId = requestPath.getOperationOrBlobId(false);
+    NamedBlobPath namedBlobPath = NamedBlobPath.parse(namedBlobId, Collections.emptyMap());
+    // This is a three steps operation
+    // 1. Get regular blob id from named blob id.
+    // 2. Delete regular blob id from Router
+    // 3. Delete database row.
+    // If we failed at step 1, or step 1 returns NotFound, then  we just return the error back to client, indicating failure
+    // If we failed at step 2, client can resend the request to ambry so we can resume the operation. If regular blob is already
+    //    deleted, Router would do a nop and return success
+    // If we failed at step 3, client can still resend the delete request to delete this blob again.
+    namedBlobDb.get(namedBlobPath.getAccountName(), namedBlobPath.getContainerName(), namedBlobPath.getBlobName())
+        .thenApply(NamedBlobRecord::getBlobId)
+        .whenComplete((blobId, exception) -> {
+          if (exception != null) {
+            callback.onCompletion(null, Utils.extractFutureExceptionCause(exception));
+            return;
+          }
+          RequestPath newRequestPath =
+              new RequestPath(requestPath.getPrefix(), requestPath.getClusterName(), requestPath.getPathAfterPrefixes(),
+                  "/ " + blobId, requestPath.getSubResource(), requestPath.getBlobSegmentIdx());
+          // Replace RequestPath in the RestRequest and call GetBlobHandler.handle.
+          restRequest.setArg(InternalKeys.REQUEST_PATH, newRequestPath);
+          try {
+            accountAndContainerInjector.injectAccountAndContainerForNamedBlob(restRequest,
```

Comment:
Like Sophie suggested [here](https://github.com/linkedin/ambry/pull/2137/files#r904272769), would it be better to check for account and container at the beginning so that we avoid network call to NamedBlob DB in case account or container is invalid.